.github/workflows/coverage-check.yml

```diff
@@ -0,0 +1,116 @@
+        run: yarn install --mode=skip-build && yarn allow-scripts
+
+      - name: Setup Foundry
+        uses: onbjerg/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Setup Forge
+        run: yarn workspace @ubiquity/contracts forge:install
+
+      - name: Get development branch coverage
+        run: |
+          # this is temporary, until we merge the pr branch into development then we can use the yarn workspaces command
+
+          mkdir packages/contracts/scripts/coverage-check && touch packages/contracts/scripts/coverage-check/coverage.ts
+
+          echo 'import { getRPC } from "../runner/rpcutil";
+          import { exec } from "child_process";
+          const setRPCURL = async () => {
+            const optimalRPC = await getRPC();
+            console.log(optimalRPC);
+            return optimalRPC;
+          };
+
+          setRPCURL();' > packages/contracts/scripts/coverage-check/coverage-check.ts    
+
+          RPC_URL=$(npx ts-node packages/contracts/scripts/coverage-check/coverage-check.ts)
+          echo $RPC_URL
+          echo "forge coverage --no-match-path=**/utils/**=**/utils/** --rpc-url $RPC_URL"
+          cd packages/contracts
+
+          COVERAGE_DEVELOPMENT_OUTPUT=$(forge coverage --no-match-path=**/utils/**=**/utils/** --rpc-url $RPC_URL)
+          COVERAGE_DEVELOPMENT=($(echo "${COVERAGE_DEVELOPMENT_OUTPUT}" | tail -n 1 | awk '{print $4,$7,$10,$13}' | tr -d '%'))
+
+          COVERAGE_DEVELOPMENT_STRING="${COVERAGE_DEVELOPMENT[0]},${COVERAGE_DEVELOPMENT[1]},${COVERAGE_DEVELOPMENT[2]},${COVERAGE_DEVELOPMENT[3]}"
+
+          echo "${COVERAGE_DEVELOPMENT_OUTPUT}"
+          echo "COVERAGE_DEVELOPMENT=${COVERAGE_DEVELOPMENT_STRING}" >> $GITHUB_ENV
+          echo "Main branch coverage - Lines: ${COVERAGE_DEVELOPMENT[0]}%, Statements: ${COVERAGE_DEVELOPMENT[1]}%, Branches: ${COVERAGE_DEVELOPMENT[2]}%, Functions: ${COVERAGE_DEVELOPMENT[3]}%"
+
+      - name: Checkout code in PR branch
+        uses: actions/checkout@v3
+
+      - name: Yarn Install
+        run: yarn install --mode=skip-build && yarn allow-scripts
+
+      - name: Get PR branch coverage
+        id: pr-coverage
+        run: |
+          yarn workspace @ubiquity/contracts forge:coverage REMOTE PR 
+          # COVERAGE_PR_OUTPUT=$(yarn workspace @ubiquity/contracts forge:coverage REMOTE)
+          # COVERAGE_PR=($(echo "${COVERAGE_PR_OUTPUT}" | tail -n 1 | awk '{print $4,$7,$10,$13}' | tr -d '%'))
+
+          # COVERAGE_PR_STRING="${COVERAGE_PR[0]},${COVERAGE_PR[1]},${COVERAGE_PR[2]},${COVERAGE_PR[3]}"
+
+          # echo "${COVERAGE_PR_OUTPUT}"
+          # echo "COVERAGE_PR=${COVERAGE_PR_STRING}" >> $GITHUB_ENV
+          # echo "PR branch coverage - Lines: ${COVERAGE_PR[0]}%, Statements: ${COVERAGE_PR[1]}%, Branches: ${COVERAGE_PR[2]}%, Functions: ${COVERAGE_PR[3]}%"
+
+      - name: Compare coverage
+        run: |
+          IFS=',' read -r -a COVERAGE_PR <<< "${{ env.COVERAGE_PR }}"
```

Comment:
what about creating a script folder and paste these scripts in it ?

Reply:
sure will do that